src/api06/changeset_upload_handler.cpp

```diff
@@ -68,6 +68,26 @@ changeset_upload_responder::changeset_upload_responder(mime::type mt,
     }
   }
 
+  if (global_settings::get_bbox_size_limiter_upload()) {
+
+    auto const cs_bbox = handler.get_bbox();
+
+    if (!(cs_bbox == bbox_t()))   // valid bbox?
+    {
+      auto const max_bbox_size = upd.get_bbox_size_limit(*user_id);
+
+      if (cs_bbox.linear_size() > max_bbox_size) {
+
+        logger::message(
+            fmt::format(
+                "Upload of {} changes by user {} in changeset {} blocked due to bbox size limit exceeded, max bbox size {}",
+                new_changes, *user_id, changeset, max_bbox_size));
+
+        throw http::payload_too_large("Upload has been blocked, since it exceeds the current bbox size limit.");
```

Comment:
I'd avoid the shorthand 'bbox' in the error message since end users might not know what that means. 

"Upload has been blocked because the affected area is too large"